os/pm/pm_wake.c

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
+#include <debug.h>
+
+#include <arch/irq.h>
+
+#include "pm.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_wakeup
+ *
+ * Description:
+ *   This function is called by a core interrupt service routine(ISR).
+ *   previously registered power management event callbacks.
+ *
+ * Input parameters:
+ *   missing_tick - Missed ticks while the core was sleeping.
+ *
+ * Returned value:
+ *   None
+ *
+ ****************************************************************************/
+
+void pm_wake(int missing_tick)
+{
+	irqstate_t flags  = irqsave();
```

Comment:
I think enter/leave_critical_section should be used here. Since we ported SMP, we are using enter/leave_critical_section. Even without SMP, enter/leave_critical_section will be redirected to irqsave/restore. Please check if a change is necessary.

Reply:
Thank you for view. Changed